websites/H/Homestuck^2/presence.ts

```diff
@@ -0,0 +1,156 @@
+const presence = new Presence({
+		clientId: "945791515169521694",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "logo",
+			startTimestamp: browsingTimestamp,
+		},
+		{ pathname } = document.location,
+		pathArr = pathname.split("/");
+
+	switch (pathArr[1]) {
+		case "":
+			presenceData.details = "Viewing the home page";
+			break;
+
+		case "story":
+			presenceData.details = "Reading Homestuck^2";
+			presenceData.smallImageKey = "heart";
+			if (!pathArr[2]) presenceData.state = `Page 1 of ${await getPages()}`;
+			else presenceData.state = `Page ${pathArr[2]} of ${await getPages()}`;
+
+			if (document.querySelector("h2"))
+				presenceData.smallImageText = document.querySelector("h2").textContent;
+			else {
+				presenceData.smallImageText =
+					document.querySelector("title").textContent;
+			}
+			presenceData.buttons = [
+				{
+					label: "Read Along",
+					url: `https://www.homestuck2.com${pathname}`,
+				},
+			];
+			break;
+
+		case "bonus":
+			presenceData.details = "Viewing bonus content";
+			switch (pathArr[2]) {
+				case "catnapped":
+					presenceData.details = "Reading Catnapped";
+					presenceData.smallImageKey = "heart";
+					presenceData.state = `Page ${pathArr[3]} of 28`;
+					presenceData.smallImageText =
+						document.querySelector("h2").textContent;
+					presenceData.buttons = [
+						{
+							label: "Read Along",
+							url: `https://www.homestuck2.com${pathname}`,
+						},
+					];
+					break;
+
+				case "a-treatise-on-representational-democracy":
+					presenceData.details =
+						"Reading A Treatise on Representational Democracy";
+					presenceData.smallImageKey = "heart";
+					presenceData.state = `Page ${pathArr[3]} of 13`;
+					presenceData.smallImageText =
+						document.querySelector("h2").textContent;
+					presenceData.buttons = [
+						{
+							label: "Read Along",
+							url: `https://www.homestuck2.com${pathname}`,
+						},
+					];
+					break;
+
+				case "diamonds-dames-and-dads":
+					presenceData.details = "Reading Diamonds, Dames, and Dads";
```

Comment:
This will be incredibly long to display on a prseence

Reply:
oh, im aware. im not really sure if its worth doing anything about though, since ive used the same naming convention for the presences of each short story on the website. its just straight up how long those particular titles are
![image](https://user-images.githubusercontent.com/38714062/180082634-147ff795-177b-4693-8859-ff249a9e3496.png)